lib/azure/armrest/configuration.rb

```diff
@@ -64,14 +64,9 @@ def self.cache_token(configuration)
       # Maximum number of threads to use within methods that use Parallel for thread pooling.
       attr_accessor :max_threads
 
-      # The environment in which to acquire your token.
-      attr_reader :environment
-
-      # The authority URL used to acquire a valid token.
-      attr_accessor :resource_url
-
-      # The resource URL used to acquire a valid token.
-      attr_accessor :authority_url
+      # The environment object which determines various endpoint URL's. The
+      # default is Azure::Armrest::Environment::Public.
+      attr_accessor :environment
 
       # Yields a new Azure::Armrest::Configuration objects. Note that you must
       # specify a client_id, client_key, tenant_id. The subscription_id is optional
```

Comment:
@bronaghs All that means is that you can call `Configuration.new` without one, but if you try to pass a `conf` object to, say, a VirtualMachineService and it isn't defined, it will fail.

In effect, this means that the subscription_id requirement is *delayed* rather than optional in most cases. Why did we do this? Because technically you don't need it to get the initial token. This was desired because we wanted to be able to get a list of subscriptions without first having to provide one.

And we wanted *that* ultimately to facilitate cloud discovery in MIQ.